src/hooks/useLockHistory.ts

```diff
@@ -1,22 +1,77 @@
-import { LockHistory } from '@/utils/boost'
-
-const FAKE_LOCKS = [
-  {
-    day: 0,
-    amount: 1000,
-  },
-  {
-    day: 10,
-    amount: 5000,
-  },
-  {
-    day: 20,
-    amount: 4000,
-  },
-]
-
-export const useLockHistory = (): LockHistory[] => {
-  return FAKE_LOCKS
+import { useAddress } from './useAddress'
+import useSWRInfinite from 'swr/infinite'
+import { useMemo } from 'react'
+import { START_TIMESTAMP } from '@/config/constants'
+import { toDaysSinceStart } from '@/utils/date'
+
+type LockingHistoryEntry =
+  | {
+      eventType: 'LOCKED'
+      executionDate: string
+      transactionHash: string
+      holder: string
+      amount: string
+      logIndex: string
+    }
+  | {
+      eventType: 'UNLOCKED'
+      executionDate: string
+      transactionHash: string
+      holder: string
+      amount: string
+      logIndex: string
+      unlockIndex: string
+    }
+  | {
+      eventType: 'WITHDRAWN'
+      executionDate: string
+      transactionHash: string
+      holder: string
+      amount: string
+      logIndex: string
+      unlockIndex: string
+    }
+
+type LockingHistoryEventPage = {
+  count: number
+  next: string | null
+  previous: string | null
+  results: LockingHistoryEntry[]
+}
+
+export const useLockHistory = () => {
+  const address = useAddress()
+
+  const getKey = useMemo(
+    () => (pageIndex: number, previousPageData: LockingHistoryEventPage) => {
+      if (!previousPageData) {
+        // Load first page
+        return `https://safe-client.staging.5afe.dev/v1/locking/${address}/history`
+      }
+      if (previousPageData && !previousPageData.next) return null // reached the end
+
+      // Load next page
+      return `https://safe-client.staging.5afe.dev/v1/locking/${address}/history?cursor=${previousPageData.next}`
```

Comment:
TODO: This needs to be chosen depending on DEV / STAGING environment, once available on prod.